perf/perf_hv_gpci.py

```diff
@@ -80,21 +93,37 @@ def run_cmd(self, cmd):
         if output.exit_status != 0:
             self.fail_cmd.append(cmd)
 
-    def test_gpci_events(self):
-        perf_stat = "perf stat"
-        perf_flags = '-C 1 -v -e'
-
-        for line in self.list_of_hv_gpci_events:
-            evt = "hv_gpci/%s,hw_chip_id=12/" % line
-            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, evt)
+    def gpci_events(self, val):
+        for line in val:
+            if line in self.list_phys:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'phys_processor_idx=?', 'phys_processor_idx=1'))
+            if line in self.list_sibling:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'sibling_part_id=?', 'sibling_part_id=2'))
+            if line in self.list_partition:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'partition_id=?', 'partition_id=1'))
+            if line in self.list_hw:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'hw_chip_id=?', 'hw_chip_id=12'))
+            if line in self.list_noid:
+                line = "%s/" % line
+
+            cmd = "perf stat -v -e %s sleep 1" % line
             self.run_cmd(cmd)
-            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, evt)
+            cmd = "perf stat --per-core -a -e %s sleep 1" % line
             self.run_cmd(cmd)
-            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, evt)
+            cmd = "perf stat --per-socket -a -e %s sleep 1" % line
             self.run_cmd(cmd)
-
         self.error_check()
 
+    def test_gpci_events(self):
+        event_list = [self.list_phys, self.list_sibling, self.list_partition,
+                      self.list_hw, self.list_noid]
+        for event in event_list:
```

Comment:
@disgoel can this be made single line like ?

`for event in [self.list_phys, self.list_sibling, self.list_partition, self.list_hw, self.list_noid]:`

Reply:
addressed requested changes

avocado run --test-runner runner perf_hv_gpci.py
JOB ID     : 46c018aa354637014df515f04da37df48537b475
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T11.07-46c018a/job.log
 (1/1) perf_hv_gpci.py:perf_hv_gpci.test_gpci_events: PASS (209.89 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T11.07-46c018a/results.html
JOB TIME   : 229.69 s